js/src/layers/Layer.js

```diff
@@ -111,6 +120,15 @@ export class LeafletLayerView extends utils.LeafletWidgetView {
       this.update_popup,
       this
     );
+    this.listenTo(
+      this.model,
+      'change:pane',
+      function() {
+        this.update_pane(),
+        this.map_view.rerender();
+      },
+      this
+    );
```

Comment:
We should probably check that this is not triggered at Map creation (so that we don't process two renders when displaying the map the first time)

Reply:
Good point, something like that is happening: when a layer asks the map to re-render (because the layer's pane has changed), I can see that this function is executed more and more. Probably because the previous layers are still hanging around somewhere?

Reply:
> Probably because the previous layers are still hanging around somewhere?

Oh yeah. We might not have been very careful in ipyleaflet about layers removing. We should probably implement proper `remove` methods in `Views`, which will remove event listeners, delete JS objects etc.

Reply:
Fixed in https://github.com/jupyter-widgets/ipyleaflet/pull/999/commits/7293dd54e35e405333b127578b6fb16223d30b63.

Reply:
But I'm calling `this.obj.remove()`, so it's on the Leaflet Map, right?

Reply:
The whole map is re-created, so the layers too. The previous layers should be garbage-collected, I guess?

Reply:
OK to merge?

Reply:
I see, ok! If there are no side-effects of calling render again then I guess it's good :) 

The readthedocs failure is unrelated